### PR TITLE
Implement dirindex decomposition model

### PR DESCRIPTION
**Pull request by anomam**
* should address #282 posted by @mikofski

### CODE REVIEW

**Comment by wholmgren** [comment]
looks great @anomam. ping me when your test is working. 

also, please go ahead and set up a new v0.4.3.txt whatsnew file and entry in the [whatsnew.rst](https://github.com/pvlib/pvlib-python/blob/master/docs/sphinx/source/whatsnew.rst) file.

**Comment by anomam** [comment]
Thanks @wholmgren . Will do!
By the way, I was wondering if you would know where I could find some (validation) data about the DIRINDEX model that could help me make sure that  my implementation is correct?

**Comment by wholmgren** [comment]
I do not have any data or a reference implementation for DIRINDEX, but maybe someone else can comment. I think it's a straightforward extension of DIRINT, though, so I'm not sure if it's really necessary.

**Comment by wholmgren** [comment]
oh, and also add your new function to the api.rst file

File: pvlib/test/test_irradiance.py [hunk 1]
```diff
@@ -365,3 +368,29 @@ def test_erbs_all_scalar():
...
+    pressure = 93193.
+    tdew = 10.
+    out = irradiance.dirindex(ghi, ghi_clearsky, dni_clearsky,
+                              zenith, times, pressure=pressure,
+                              temp_dew=tdew)
+    dirint_close_values = irradiance.dirint(ghi, zenith, times,
+                                            pressure=pressure,
+                                            use_delta_kt_prime=True,
+                                            temp_dew=tdew).values
+    expected_out = np.array([np.nan, 0., 748.31562753, 630.72592644])
```
Comment (wholmgren) [inline pvlib/test/test_irradiance.py]: Does this nan arise from 0/0? If so, I think most users would probably prefer to get a 0 here instead. Something like ``dni_dirindex[(dni_dirindex < 0) | dni_dirindex.isnan()] = 0.`` should work.

File: pvlib/test/test_irradiance.py [hunk 2]
```diff
@@ -365,3 +368,29 @@ def test_erbs_all_scalar():
...
         assert_allclose(v, expected[k], 5)
+
+
+def test_dirindex():
+    clearsky_data = tus.get_clearsky(times, model='ineichen',
+                                     linke_turbidity=3)
+    ghi = pd.Series([0, 0, 1038.62, 254.53], index=times)
+    ghi_clearsky = clearsky_data['ghi']
+    dni_clearsky = clearsky_data['dni']
+    zenith = ephem_data['apparent_zenith']
```
Comment (wholmgren) [inline pvlib/test/test_irradiance.py]: Your test is going to be more robust you hard code these input values. The values themselves are fine, it's only that they're derived at runtime from functions that could produce slightly different results in the future and thus might make your new test fail for no good reason. You'll can see that the inputs are precisely defined in the dirint tests that actually check values.

Comment (anomam) [inline pvlib/test/test_irradiance.py]: Got it!

File: pvlib/test/test_irradiance.py (hunk 1 above)
Comment (anomam) [inline pvlib/test/test_irradiance.py]: It comes from the relative airmass function and its use in the dirint model from what I saw. I think that ``airmass`` returns ``NaN`` for night times.

Comment (wholmgren) [inline pvlib/test/test_irradiance.py]: Oh right, thanks for reminding me. My suggestion was bad anyways since it would have yielded 0s for any nan inputs.

**Pull request merged**